src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalTestDetectors.cs

```diff
@@ -85,10 +97,29 @@ public static bool IsServerLocalHost()
             return index == 0;
         }
 
-        public static bool AreUserNameAndPasswordAvailable()
+        // Returns 'true' if ambient credentials are avaiable to use
+        public static bool AreAmbientCredentialsAvailable()
         {
-            return !String.IsNullOrWhiteSpace(TestProperties.GetProperty(TestProperties.TestUserName_PropertyName)) &&
-                   !String.IsNullOrWhiteSpace(TestProperties.GetProperty(TestProperties.TestPassword_PropertyName));
+            // Requires solution to https://github.com/dotnet/wcf/issues/1095
+            // Till then, heuristic is that running localhost == ambient credentials available
+            return IsServerLocalHost();
+        }
+
+        // Returns 'true' if explicit credentials are available.
+        // Currently, 'true' means the TestProperties for ExplicitUserName
+        // and ExplicitPassword are non-blank.
+        public static bool AreExplicitCredentialsAvailable()
+        {
+            return !String.IsNullOrWhiteSpace(TestProperties.GetProperty(TestProperties.ExplicitUserName_PropertyName)) &&
+                   !String.IsNullOrWhiteSpace(TestProperties.GetProperty(TestProperties.ExplicitPassword_PropertyName));
```

Comment:
The intent was that both would need to be non-blank for explicit credentials.  Currently these are used only by the Digest test and need to be manually specified.  If you think it is better to allow for a blank password, I can get purely off the ExplictUserName.  It would not be legal to have a password with no user name.  Thoughts welcome.